aif360/metrics/binary_label_dataset_metric.py

```diff
@@ -26,13 +27,28 @@ def __init__(self, dataset, unprivileged_groups=None, privileged_groups=None):
             TypeError: `dataset` must be a
                 :obj:`~aif360.datasets.BinaryLabelDataset` type.
         """
-        if not isinstance(dataset, BinaryLabelDataset):
-            raise TypeError("'dataset' should be a BinaryLabelDataset")
+        if not isinstance(dataset, BinaryLabelDataset) and not isinstance(dataset, MulticlassLabelDataset) :
+            raise TypeError("'dataset' should be a BinaryLabelDataset or a MulticlassLabelDataset")
 
         # sets self.dataset, self.unprivileged_groups, self.privileged_groups
         super(BinaryLabelDatasetMetric, self).__init__(dataset,
             unprivileged_groups=unprivileged_groups,
             privileged_groups=privileged_groups)
+        
+        if isinstance(dataset, MulticlassLabelDataset):
+            fav_label_value = 1.
+            unfav_label_value = 0.
+            #Iterate the labels values in the dataframe and check whether the label value is 
+            # in the favorable list, if it is true then assign the favorable label value as 1
+            # otherwise 0 for unfavourable values
+            for index in range(0,len(self.dataset.labels)):
+                if self.dataset.labels[index] in self.dataset.favorable_label:
+                    self.dataset.labels[index] = float(fav_label_value)
+                elif self.dataset.labels[index] in self.dataset.unfavorable_label:
+                    self.dataset.labels[index] = float(unfav_label_value)
```

Comment:
This will modify the original dataset which we don't want to do. We should make a copy first and then change the labels, i.e.:
```py
self.dataset = self.dataset.copy()
# find all labels which match any of the favorable labels
fav_idx = np.logical_or.reduce(np.equal.outer(self.dataset.labels, self.dataset.favorable_label))
# replace labels with corresponding values
self.dataset.labels = np.where(fav_idx, fav_label_value, unfav_label_value)
```

Reply:
Thanks Sam. Updated the code as per the suggestion.